Firestore/core/src/firebase/firestore/core/sync_engine.cc

```diff
@@ -367,11 +373,18 @@ DocumentKeySet SyncEngine::GetRemoteKeys(TargetId target_id) const {
       it->second.document_received) {
     return DocumentKeySet{it->second.key};
   } else {
-    auto found = query_views_by_target_.find(target_id);
-    if (found != query_views_by_target_.end()) {
-      return found->second->view().synced_documents();
+    DocumentKeySet keys;
+    if (queries_by_target_.count(target_id) == 0) {
+      return keys;
+    }
+
+    for (auto& query : queries_by_target_.at(target_id)) {
```

Comment:
Since we don't need or want to mutate the query here (or the key below) these should be `const auto&`.

Reply:
Done.

Reply:
Please make both `query` and `key` `const auto&`

Reply:
Done.